h2o-r/tests/testdir_algos/targetencoder/runit_targetencoder_model.R

```diff
@@ -0,0 +1,26 @@
+setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
+source("../../../scripts/h2o-r-test-setup.R")
+
+test.model.targetencoder <- function() {
+    data <- h2o.importFile(path = locate('smalldata/gbm_test/titanic.csv'), col.types=list(by.col.name=c("survived"),types=c("factor")))
+    encoded_columns <- c('home.dest', 'cabin', 'embarked')
+    target_encoder <- h2o.targetencoder(training_frame = data, encoded_columns= encoded_columns, target_column = "survived")
+    encoded_data <- h2o.transform(target_encoder, data) # For now, there is only the predict method
+    expect_false(is.null(encoded_data))
+    expect_equal(h2o.ncol(data) + length(encoded_columns), h2o.ncol(encoded_data))
+    expect_true(h2o.nrow(data) == 1309)
+    
+    # Test fold_column proper handling + kfold data leakage strategy defined
+    target_encoder <- h2o.targetencoder(training_frame = data, encoded_columns= encoded_columns, target_column = "survived",
```

Comment:
We now have three different TE interfaces in R... @meganjkurka's original, @deil87's recent one and now the `h2o.targetencoder()`.  @Pscheidl I assume making a new API is the only way to achieve what you're trying to do?

Reply:
True. In history, there were other attempts to bypass model auto-generated API due to various reasons. Word2Vec was one of them, TargetEncoder is the most recent API.

This PR, besides integrating TE as a Model, it also paves the way (not only by introducing a delegating `transform method, but also by many enhancements on our backend) for future transformers of all kinds to be usable as models. It will also serve as a showcase for others on how to re-use what we already have. This is **what I tried to achieve**, together with unification of the APIs in future.

Therefore, if I want to unify current and future models under one API and reuse current ecosystem at the same time, avoiding the long journey it takes to develop, test and polish new API, there is no other way to me. If there was, I'd take it.